pkg/kclient/kclient.go

```diff
@@ -228,3 +233,38 @@ func (c *Client) IsResourceSupported(apiGroup, apiVersion, resourceName string)
 	}
 	return supported, nil
 }
+
+// IsSSASupported checks if Server Side Apply is supported by cluster
+// SSA was introduced in Kubernetes 1.16
+// If there is an error while parsing versions, it assumes that SSA is supported by cluster.
+// Most of clusters these days are 1.16 and up
+func (c *Client) IsSSASupported() bool {
+	// check if this was done before so we don't query cluster multiple times for the same info
+	if c.isSSASupported == nil {
+		versionWithSSA, err := semver.Make("1.16.0")
+		if err != nil {
+			klog.Warningf("unable to parse version %q", err)
```

Comment:
Yes, good catch, I will add return.
The reason I added this check was to make sure that if in the future someone updates the `1.16.0` string and makes a typo it will be counght.

Reply:
In this case, you can perhaps make an assertion and panic, like it is done for `regexp.MustCompile` for example (https://golang.org/pkg/regexp/#MustCompile), so the developer will see the error quickly if it makes a typo